spec/ddtrace/contrib/active_storage/tracer_spec.rb

```diff
@@ -0,0 +1,195 @@
+require 'ddtrace/contrib/support/spec_helper'
+require 'ddtrace/contrib/analytics_examples'
+require 'ddtrace/contrib/integration_examples'
+require 'ddtrace/contrib/rails/rails_helper'
+require 'ddtrace'
+require 'sidekiq/testing'
+
+# we use these classes to help mock active storage
+require 'active_storage/engine'
+require 'active_storage/attached'
+
+if PlatformHelpers.jruby?
+  require 'activerecord-jdbc-adapter'
+else
+  require 'mysql2'
+  require 'sqlite3'
+end
+
+RSpec.describe 'ActiveStorage instrumentation' do
+  include Rack::Test::Methods
+  include_context 'Rails test application'
+
+  Sidekiq.configure_client do |config|
+    config.redis = { url: ENV['REDIS_URL'] }
+  end
+
+  Sidekiq.configure_server do |config|
+    config.redis = { url: ENV['REDIS_URL'] }
+  end
+
+  Sidekiq::Testing.inline!
+
+  let(:mysql) do
+    {
+      database: ENV.fetch('TEST_MYSQL_DB', 'mysql'),
+      host: ENV.fetch('TEST_MYSQL_HOST', '127.0.0.1'),
+      password: ENV.fetch('TEST_MYSQL_ROOT_PASSWORD', 'root'),
+      port: ENV.fetch('TEST_MYSQL_PORT', '3306')
+    }
+  end
+
+  def mysql_connection_string
+    "mysql2://root:#{mysql[:password]}@#{mysql[:host]}:#{mysql[:port]}/#{mysql[:database]}"
+  end
+
+  let(:application_record) do
+    stub_const('ApplicationRecord', Class.new(ActiveRecord::Base) do
+      self.abstract_class = true
+    end)
+
+    stub_const('Article', Class.new(ApplicationRecord) do
+      # active storage initializers have changed from 5 => 6 so we need to mimic both approaches
```

Comment:
We have different application files for different versions of Rails: https://github.com/DataDog/dd-trace-rb/tree/master/test/contrib/rails/apps

Do you think it makes sense to move this to `rails5.rb` and `rails6.rb` respectively?

It could be something like declaring a global method, like:
```ruby
stub_const('Article', new_application_record)
```

or a mixin:
```ruby
stub_const('Article', Class.new(ApplicationRecord) do
	include Datadog::Testing::ApplicationRecord
end)
```

What do you think?

Reply:
makes sense, can clean this up